readthedocs/core/views/__init__.py

```diff
@@ -116,7 +116,9 @@ def server_error_404(request, exception=None, template_name='404.html'):  # pyli
         Marking exception as optional to make /404/ testing page to work.
     """
     response = get_redirect_response(request, path=request.get_full_path())
-    if response:
+
+    if response and response.url != request.build_absolute_uri():
```

Comment:
> I wonder if we should add some logging here if there is a loop

:+1: on adding a `log.warning` if this happens.

> Perhaps we should show a different error code in this case, perhaps a 500 level, to show that something is broken vs. just missing?

Mmm... Well, for the use case were we detected this, what we want here is a 404 not a 500 actually (you can read the issue that this PR closes for more context).

I suppose that returning a 404 is the first step here to avoid this problem.

Reply:
Yea, better than actually returning an infinite redirect. So I'm +1 on shipping and then improving later.